File: CS50x/PSET2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    //Getting a text from user

    string text = get_string ("Text: ");

    //Calculating the letters, words and sentences

    int numC = 0;
    int numW = 1;
    int numS = 0;

    for(int i =0; i < strlen(text); i++)
    {
        //Calculating the letters

        if(('a' <= text[i] && text[i] <= 'z') || ('A' <= text[i] && text[i] <= 'Z'))
        {
            numC++;
        }

        //Calculating the words

        int w = text[i];
        if(w == 32)
        {
            numW++;
        }

        //Calculating the sentences

        if(w == 46 || w == 33 || w == 63 )
        {
            numS++;
        }
    }

    //Calculating the grade

    int L = numC*100/numW;
    int S = numS*100/numW;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int rounded = round(index);

    //Printing the grade

    if (rounded >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (rounded < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", rounded);
    }

}

File: CS50x/PSET6/sentimental-readability/readability.py
# Getting a text from user
text = str(input("Text: "))

# Calculating the letters, words and sentences
c = 0
for i in text:
    if 'a' <= i <= 'z' or 'A' <= i <= 'Z':
        c += 1

w = 1
for i in text:
    if i == " ":
        w += 1

s = 0
for i in text:
    if i == "." or i == "!" or i == "?":
        s += 1

# Calculating the grade
L = float(c * 100 / w)
S = float(s * 100 / w)
index = round(0.0588 * L - 0.296 * S - 15.8)

# Printing the grade
if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")